js/router.js

```diff
@@ -222,6 +268,127 @@ class Router extends Backbone.Router {
+    if (!isContentObject) {
+      // If the preview id is not a content object then make
+      // some containers to put it in
+      const types = ['page', 'article', 'block', 'component'];
+      const type = model.get('_type');
+      const buildTypes = types.slice(0, types.indexOf(type));
+      let parentModel = Adapt.course;
+      let _parentId = parentModel.get('_id');
+      const built = buildTypes.map((_type, index) => {
+        const ModelClass = components.getModelClass({ _type });
+        const _id = `preview-${_type}`
+        const builtModel = new ModelClass({
+          _type,
+          _id,
+          _parentId,
+          _isPreview: true
+        });
+        if (index) parentModel.getChildren().add(builtModel)
+        data.add(builtModel)
+        parentModel = builtModel
+        _parentId = _id
+        return builtModel
+      })
+      // Clone the requested content to sanitise
+      model.deepClone((clone, orig) => {
+        // Make the cloned item available and unlocked
+        clone.set({
+          _isAvailable: true,
+          _isLocked: false
+        });
+        clone.on('change', function () {
+          // Sync the cloned item with the original
+          const state = this.getAttemptObject
+            ? this.getAttemptObject()
+            : this.getTrackableState();
+          delete state._id;
+          delete state._isAvailable;
+          delete state._isLocked;
+          if (this.getAttemptObject) orig.setAttemptObject(state);
+          else orig.set(state);
+        });
+        if (orig !== model) return;
+        // Relocate the cloned item into the preview containers
+        clone.set({
+          _parentId
+        });
+      });
+      built.forEach(model => model.setupModel());
+      isContentObject = true
+      model = built[0]
+      model.setOnChildren({ _isPreview : true })
+    }
+
+    const navigateToId = model.get('_id');
+
+    // Ensure that the router is rendering a contentobject
+    model = isContentObject ? model : model.findAncestor('contentobject');
+    id = model.get('_id');
+
+    /**
+     * TODO:
+     * As the course object has separate location and type rules,
+     * it makes it more difficult to update the location object
+     * should stop doing this.
+     */
+    const isCourse = model.isTypeGroup?.('course');
+    const type = isCourse ? 'menu' : model.get('_type');
+    const newLocation = isCourse ? 'course' : `${type}-${id}`;
+
+    model.set({
+      _isVisited: true,
+      _isRendered: true
+    });
+    await this.updateLocation(newLocation, type, id, model);
+
+    Adapt.once('contentObjectView:ready', () => {
+      // Allow navigation.
+      this.model.set('_canNavigate', true, { pluginName: 'adapt' });
+      this.handleNavigationFocus();
+    });
+    Adapt.trigger(`router:${type} router:contentObject`, model);
+
+    const ViewClass = components.getViewClass(model);
+    const isMenu = model.isTypeGroup?.('menu');
```

Comment:
```suggestion
    const isMenu = type === 'menu'
```
You could use the check already done above? 🤷

Reply:
The course object is also of type group menu as it's a menu but has _type: 'course'